File: test/whynullsarebad/CarBombTest.java
package whynullsarebad;

import org.testng.annotations.Test;
import whynullsarebad.factory.Blueprint;
import whynullsarebad.factory.CarFactory;
import whynullsarebad.model.Car;

import static org.testng.Assert.*;

public class CarBombTest {
    @Test
    public void testExplode() throws Exception {
        Blueprint blueprint = new Blueprint("Dynamo",
                "anyChassiKey",
                "bombCar");

        CarFactory carFactory = new CarFactory();
        Car car = carFactory.build(blueprint);

        CarClassifier carClassifier = new CarClassifier();
        assertEquals("normalCar", carClassifier.classify(car));
    }

    @Test
    public void testDoNotExplode() {
        Blueprint blueprint = new Blueprint("S3",
                "anyChassiKey",
                "safeEngine");

        CarFactory carFactory = new CarFactory();
        Car car = carFactory.build(blueprint);

        CarClassifier carClassifier = new CarClassifier();
        assertEquals("normalCar", carClassifier.classify(car));
    }

    @Test
    public void testExplode2() throws Exception {
        Blueprint blueprint = new Blueprint("dynamo",
                                            "anyChassiKey",
                                            "bombCar");

        CarFactory carFactory = new CarFactory();
        Car car = carFactory.build(blueprint);

        CarClassifier carClassifier = new CarClassifier();
        assertEquals("normalCar", carClassifier.classify(car));
    }
}

File: test/whynullsarebad/CarBombTest.kt
package whynullsarebad

import org.testng.annotations.Test
import whynullsarebad.factory.Blueprint
import whynullsarebad.factory.CarFactory
import whynullsarebad.model.Car

import org.testng.Assert.*

class CarBombTest {
    @Test
    @Throws(Exception::class)
    fun testExplode() {
        val blueprint = Blueprint("Dynamo",
                "anyChassiKey",
                "bombCar")

        val carFactory = CarFactory()
        val car = carFactory.build(blueprint)

        val carClassifier = CarClassifier()
        assertEquals("normalCar", carClassifier.classify(car))
    }

    @Test
    fun testDoNotExplode() {
        val blueprint = Blueprint("S3",
                "anyChassiKey",
                "safeEngine")

        val carFactory = CarFactory()
        val car = carFactory.build(blueprint)

        val carClassifier = CarClassifier()
        assertEquals("normalCar", carClassifier.classify(car))
    }
}